File: movies_picker_client/lib/data/models/language.dart
import '../../shared/shared.dart';

enum Language {
  english,
  hindi,
}

extension LanguageExtension on Language {
  String get display => SharedFunctions.getEnumName(this);
}

abstract class LanguageParser {
  static Language parse(String s) {
    switch (s) {
      case 'ENGLISH':
        return Language.english;
      case 'HINDI':
        return Language.hindi;
      default:
        return Language.english;
    }
  }
}

File: movies_picker_server/movies-picker/src/main/java/moviespicker/model/Language.java
package moviespicker.model;

public enum Language {
    ENGLISH,
    HINDI;

    public static Language parse(String s) {
        switch (s.toUpperCase()) {
            case "HINDI": return HINDI;
            case "ENGLISH":
            default:
                return ENGLISH;
        }
    }
}
